lib/crypto/crypto_scrypt_smix_sse2.c

```diff
@@ -166,9 +166,12 @@ blockmix_salsa8(const __m128i * Bin, __m128i * Bout, __m128i * X, size_t r)
 static uint64_t
 integerify(const __m128i * B, size_t r)
 {
-	const uint32_t * X = (const uint32_t *)(B + (2 * r - 1) * 4);
+	const __m128i * X = B + (2*r - 1) * 4;
+	uint32_t X0, X13;
 
-	return (((uint64_t)(X[13]) << 32) + X[0]);
+	X0 = (uint32_t)_mm_cvtsi128_si32(X[0]);
+	X13 = (uint32_t)_mm_cvtsi128_si32(X[13]);
```

Comment:
X is an array of 4-element vectors.  So we want the first element of X[0] and the second element of X[3].